bp_top/test/common/dromajo_cosim.cpp

```diff
@@ -28,8 +28,8 @@ extern "C" void dromajo_init(char* cfg_f_name, int hartid, int ncpus, int memory
       dromajo_pointer = dromajo_cosim_init(6, argv);
     }
     else {
-      char* argv[] = {"dromajo", (char*)(&ncpus_str[0]), (char*)(&memsize_str[0]), (char*)(&mmio_str[0]), bootrom_str, "prog.elf"};
```

Comment:
I think this breaks checkpointing? @farzamgl ?

Reply:
No, the checkpoint option is in the if clause.

Reply:
Ah I see.  So dromajo is just using the default bootrom in all cases then?

@sripathi-muralitharan delete the bootrom_str variable in this file and the bootrom.bin copying in the makefile flow, if it's unused.

Reply:
Well shouldn't we keep it because we are going to add a BOOTCHECK flag later on right?

Reply:
It's using the *.boorom generated file in the checkpointing case. So yes we don't need the --bootrom option.

Reply:
> It's using the *.bootrom generated file in the checkpointing case. So yes we don't need the --bootrom option.

Got it.

BOOTCHECK is now (this PR) or never.  I lean towards never, since any errors in bootrom will be pretty easily detectable.  We could also add a test where we generate a bootrom with arbitrary data, and then the program is literally "add x1, x0; add x2, x0... csrr mhartid, csrr mcause; ...."?  The weirdness would be setting a non-default bootrom in the flow, not sure if it's worth the headache

Reply:
Yes, let's just focus on the new boot flow. It's probably difficult maintaining both of them.

Reply:
>@sripathi-muralitharan delete the bootrom_str variable in this file and the bootrom.bin copying in the makefile flow, if it's unused.

Did not remove the bootrom.bin since its used to create the bootrom.mem required to initialize the nonsynth_rom (bootrom) in the host.

Reply:
You can go directly from riscv to mem as seen here: https://github.com/black-parrot/black-parrot/blob/master/bp_common/test/Makefile.tests#L82.  We only went from the binary because it was already available

Reply:
Oh alright. Added another commit with this change. Now bootrom.bin files are not generated as part of the flow.